granadilla/forms.py

```diff
@@ -74,3 +75,31 @@ def save(self, commit=True):
     class Meta:
         model = models.LdapUser
         fields = ('phone', 'mobile_phone', 'internal_phone', 'new_photo')
+
+
+class LdapUserPassForm(forms.Form):
+    current_pass = forms.CharField(label='Current password', max_length=50, widget=forms.PasswordInput)
+    new_pass_1 = forms.CharField(label='New password', max_length=50, widget=forms.PasswordInput)
+    new_pass_2 = forms.CharField(label='New password again', max_length=50, widget=forms.PasswordInput)
+
+    def clean(self):
+        cleaned_data = super(LdapUserPassForm, self).clean()
+        new_pass_1 = cleaned_data.get("new_pass_1")
+        new_pass_2 = cleaned_data.get("new_pass_2")
+        if new_pass_2 is not None:
+            if new_pass_1 != new_pass_2:
+                raise forms.ValidationError("New password is not the same!")
+            if zxcvbn.zxcvbn(self.cleaned_data['new_pass_2'])['score'] < 4:
+                raise forms.ValidationError("Your password too weak!")
+            if not self.user.check_password(self.cleaned_data['current_pass']):
+                raise forms.ValidationError("Invalid Password")
+        return cleaned_data
+
+    def save(self, commit=True):
+        self.user.set_password(self.cleaned_data['new_pass_2'])
+        self.user.save()
+        return self.user
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
```

Comment:
Since the ``user`` parameter is required, you could use the (Python3-based) syntax:

```python
def __init__(self, *args, *, user, **kwargs):
    self.user = user
    super().__init__(*args, **kwargs)
```

This will force the caller to always provide ``user=X`` as a keyword argument.